### PR TITLE
#3236 Added dev/config view

### PR DESCRIPTION
**Pull request by jakxnz**
Added a dev view to output the current config from `dev/config`

e.g `.vendor/silverstripe/framework/sake dev/config` or `/dev/config` in a browser

Output is a simple Yaml representation of the config manifest
image

https://github.com/silverstripe/silverstripe-framework/issues/3236

UPDATE:

Added a dev view to output missing property definitions in the current config from `dev/config/audit`

image

### CODE REVIEW

**Comment by jakxnz** [comment]
Sorry for the staggered feature additions.

I've added `config/audit` for matching missing property definitions in the current config

**Comment by dhensby** [comment]
Just a quick question, which is mostly because I've forgotten, but how are the view permissions enforced on this endpoint?

**Comment by lerni** [comment]
This feature has the potential, that I finally split up my config files adequately ;)

**Comment by jakxnz** [comment]
@dhensby in `live` environment types, users must have permission and are also intercepted with a "confirm" screen before viewing. In `dev` there are no view restrictions

File: src/Dev/DevConfigController.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Config\Config;
+use Symfony\Component\Yaml\Yaml;
+use SilverStripe\Core\Injector\Injector;
+
+/**
+ * Class DevConfigController
+ *
+ * @package SilverStripe\Dev
```
Comment (maxime-rainville) [inline src/Dev/DevConfigController.php:16]:
```suggestion
 * Outputs the full configuration.
```

File: src/Dev/DevConfigController.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    /**
+     * Returns all the keys of a multi-dimensional array while maintining any nested structure
+     *
+     * @param array $array
+     * @param int $maxdepth
+     * @param int $depth
+     * @param array $arrayKeys
+     * @return array
+     */
+    protected function array_keys_recursive($array, $maxdepth = 20, $depth = 0, $arrayKeys = [])
```
Comment (maxime-rainville) [inline src/Dev/DevConfigController.php:143]:
```suggestion
    private function array_keys_recursive($array, $maxdepth = 20, $depth = 0, $arrayKeys = [])
```

Unless we actually expect people to extend this controller, I think we should make this method private.

**Review by chillu — APPROVED** [review]

**Pull request merged**